geps/gep-1709.md

```diff
@@ -253,6 +253,44 @@ move on to [certification](#certification) to report the results.
 [go]:https://go.dev
 [lib]:https://pkg.go.dev/sigs.k8s.io/gateway-api@v0.6.2/conformance/utils/suite
 
+### Gateway API version and channel
+
+The certification is related to a specific API version and a specific channel,
+therefore such information must be included in the final report.
+At test suite setup time, the conformance profile
+machinery gets all the CRDs with the field `.spec.group` equal to
+`gateway.networking.k8s.io`, and for each of them checks the annotations
+`gateway.networking.k8s.io/bundle-version` and
+`gateway.networking.k8s.io/channel`. If there are `CRD`s with different
+versions, the certification fails specifying that it's not possible to run the
+tests as there are different Gateway API versions installed in the cluster. If
+there are CRDs with different channels, the certification fails specifying
+that it's not possible to run the tests as there are different Gateway API
+channels installed in the cluster. If all the Gateway API `CRD`s have the same
+version and the same channel, the tests can be run and the detected version
+and channel will be set in the `GatewayAPIVersion` and `gatewayAPIChannel`
+fields of the final report. Furthermore, the suite must run all the
+experimental tests when the channel is `experimental`, and the related features
+are enabled.
+
+In addition to the `CRD`s version, the suite needs to check its version in
+relation to the `CRD`s one. To do so, a new `.go` file containing the current
```

Comment:
What happens when there's some skew here? For example, most of our patch releases just include new conformance tests without meaningful changes to the API spec. Would these not run if the CRD version was v0.7.0 and the test version was either v0.7.1 or v0.7.0-dev?

Reply:
That makes sense, we could allow a skew within the same minor version and block conformance where the CRDs and the suite have different minor versions.

Reply:
Is this a situation where we should start as strict as possible, and ease off _if_ we get complaints?

Reply:
I think that's also fine. I guess this doesn't prevent running conformance tests, just generating a conformance report, right? I think it's fine to limit conformance reports to tightly matching versions, but I don't want to prevent running the latest set of conformance tests based on this.

Reply:
Yep, running conformance tests won't be affected, only generating reports.